### PR TITLE
Remove audit log formatter and the history endpoint

### PR DESCRIPTION
**Pull request by cyli**
Since this is all going to be done via cloud feeds now.

### CODE REVIEW

**Comment by cyli** [comment]
jenkins please run integration tests

**Comment by cyli** [comment]
Since we don't actually use apiary, nor do we seem to keep that file up to date, wondering if we should remove it as well?

**Comment by cyli** [comment]
jenkins please run integration tests

**Comment by manishtomar** [comment]
Sorry @cyli you had put in so much work on this :(

**Comment by manishtomar** [comment]
Could you run this on devvm and do a sanity check if all the logs are printed on console?

**Comment by cyli** [comment]
@manishtomar Nah, removal of dead code is always great.  :D  And we don't have to maintain CF infra, which is also great.  And will do, great suggestion!

**Comment by wirehead** [comment]
I keep wanting to dig up the api blueprint and see if I can make it more reasonable using the latest apiary features.

File: otter/log/formatters.py [hunk 1]
```diff
@@ -332,11 +244,8 @@ def Observer(eventDict):
             if key not in PRIMITIVE_FIELDS:
                 log_params[key] = value
 
-        # emit an audit log entry also, if it's an audit log
         if 'audit_log' in eventDict:
```
Comment (radix) [inline otter/log/formatters.py]: Can't you delete this whole conditional?

Comment (cyli) [inline otter/log/formatters.py]: I think we use this in our dashboards for now

**Comment by cyli** [comment]
Sanity check for logs - a quick visual examination didn't show anything that interesting, but it's hard to see any differences looking at a giant series of logs.  So I ran the cafe policies and integration tests for the worker for this branch and master, and used the following script to analyze the resultant logs:

```
"""Assuming logs from the debug logger"""
import json
import re
import sys

from collections import defaultdict


filename = sys.argv[1]

with open(filename) as f:
    data = f.read()

data = re.sub("^2015-\d+-\d+_\d+:\d+:\d+\.\d+ ", "", data, flags=re.M)
data = re.sub("^\}$", "},", data, flags=re.M)
data = data.lstrip('[').rstrip(']').strip().rstrip(',').strip()
data = '[\n' + data + '\n]'

with open(filename, 'wb') as f:
    f.write(data)

logs = json.loads(data)

logs = [log for log in logs if (
        log.get('otter_facility') != 'kazoo' and
        "Partition allocating" not in log['message'] and
        'buckets' not in log and
        "Starting factory" not in log['message'] and
        "Stopping factory" not in log['message'] and
        log.get('category') != 'locking'
        )]

print "num logs: ", len(logs)
audit_logs = [log for log in logs if log.get('audit_log_event_source')]
print "num audit_log_sources: ", len(audit_logs)
print "num audit_logs:", len([log for log in logs if log.get('audit_log')])
print "num traceback logs: ", len([
    log for log in logs if "Traceback" in json.dumps(log)])
print "num formatting errors: ", len(
    [log for log in logs if log.get('message_formatting_error')])

by_event_type = defaultdict(int)
by_facility = defaultdict(int)

for log in logs:
    if log.get("audit_log_event_source"):
        by_event_type[log['event_type']] += 1
    if "otter_facility" in log:
        by_facility[log['otter_facility']] += 1

print json.dumps(by_event_type, indent=2)
print json.dumps(by_facility, indent=2)

with open(filename, 'wb') as f:
    json.dump(logs, f, indent=2)
```


## Worker
- Master:

        num logs:  18137
        num audit_log_sources:  673
        num audit_logs: 673
        num traceback logs:  25
        num formatting errors:  18
        {
          "server.delete": 188, 
          "server.deletable": 60, 
          "request.policy.delete": 1, 
          "convergence.scale_down": 83, 
          "server.active": 128, 
          "convergence.scale_up": 111, 
          "request.policy.create": 102
        }
        {
          "otter.rest.policies.execute_policy": 496, 
          "otter.rest.configs.edit_config_for_scaling_group": 320, 
          "otter.rest.policies.get_policy": 18, 
          "otter.silverberg": 3962, 
          "otter.rest.policies.delete_policy": 3, 
          "otter.rest.configs.edit_launch_config": 8, 
          "otter.rest.groups.get_scaling_group_state": 2618, 
          "otter.job.delete": 1400, 
          "treq.request": 1990, 
          "otter.auth.cache": 860, 
          "otter.rest.groups.create_new_scaling_group": 421, 
          "otter.rest.webhooks.execute_webhook": 91, 
          "otter.supervisor.validate_launch_config": 130, 
          "otter.rest.groups.delete_scaling_group": 122, 
          "otter.rest.policies.create_policies": 306, 
          "otter.job.launch": 1382, 
          "otter.rest.policies.update_policy": 22, 
          "CassScalingGroup": 255, 
          "otter.rest.webhooks.create_webhooks": 148, 
          "otter.rest.configs.view_launch_config": 734, 
          "otter": 2178
        }


- This branch:

        num logs:  17416
        num audit_log_sources:  673
        num audit_logs: 0
        num traceback logs:  25
        num formatting errors:  18
        {
          "server.delete": 188, 
          "server.deletable": 59, 
          "request.policy.delete": 1, 
          "convergence.scale_down": 83, 
          "server.active": 129, 
          "convergence.scale_up": 111, 
          "request.policy.create": 102
        }
        {
          "otter.rest.policies.execute_policy": 496, 
          "otter.rest.configs.edit_config_for_scaling_group": 320, 
          "otter.rest.policies.get_policy": 18, 
          "otter.silverberg": 3953, 
          "otter.rest.policies.delete_policy": 3, 
          "otter.rest.configs.edit_launch_config": 8, 
          "otter.rest.groups.get_scaling_group_state": 2594, 
          "otter.job.delete": 1400, 
          "treq.request": 1990, 
          "otter.auth.cache": 860, 
          "otter.rest.groups.create_new_scaling_group": 421, 
          "otter.rest.webhooks.execute_webhook": 91, 
          "otter.supervisor.validate_launch_config": 130, 
          "otter.rest.groups.delete_scaling_group": 122, 
          "otter.rest.policies.create_policies": 306, 
          "otter.job.launch": 1382, 
          "otter.rest.policies.update_policy": 22, 
          "CassScalingGroup": 255, 
          "otter.rest.webhooks.create_webhooks": 148, 
          "otter.rest.configs.view_launch_config": 732, 
          "otter": 2165
        }


There are the same number of audit log sources in both, the discrepancy in the number of total logs between master and this branch are explained by:

1. The extra 673 audit log=True logs that have been removed in this branch but are in master
1. A few extra pollings to view state by the tests
1. Slight timing differences

**Comment by manishtomar** [comment]
@cyli That is a lot of work w.r.t sanity check :). I was just thinking of glancing at the logs and see if you notice any glaring errors. Sorry, if I made you think otherwise.

**Comment by cyli** [comment]
@manishtomar Eh, this made me slightly more confident that I wasn't breaking anything.  And I can also use this script again to see if adding copy to the observer chain as suggested by #1580 will help with the number of formatting errors.

**Comment by cyli** [comment]
@manishtomar I also did a very basic visual inspection to see that the format some of the existing logs looked similar - I didn't look very hard, but likely this change shouldn't do much to those.

File: otter/log/formatters.py (hunk 1 above)
Comment (manishtomar) [inline otter/log/formatters.py]: Can we change the dashboard and remove this altogether before merging?

Comment (cyli) [inline otter/log/formatters.py]: There is one dashboard specifically of audit log feeds, which I changed to use this one.  I'm not sure if it's used by support at all atm.  Also we have way too many dashboards to even list, so I can't be sure I got all of them.

Comment (cyli) [inline otter/log/formatters.py]: Although hmm... maybe I can figure out what the specifically elasticsearch query is to list all dashboards as JSON, and search that way.  Kibana 3 limits the number displayed, doesn't actually create a DOM element for every one and hide them - it just creates a set number and hides/repopulates them.

Although there is still that one audit log dashboard.

Comment (cyli) [inline otter/log/formatters.py]: Ok, done.  :D

Comment (manishtomar) [inline otter/log/formatters.py]: Yay @cyli . You are the best :)

**Comment by radix** [comment]
LGTM :)

**Pull request merged**